plugin/evm/syncervm_client.go

```diff
@@ -330,12 +344,14 @@ func (client *stateSyncerClient) finishSync() error {
 	if !ok {
 		return fmt.Errorf("could not convert block(%T) to *chain.BlockWrapper", wrapper)
 	}
-	evmBlock, ok := wrapper.Block.(*Block)
+	wrappedBlock := wrapper.Block
+
+	evmBlockGetter, ok := wrappedBlock.(EthBlockWrapper)
 	if !ok {
-		return fmt.Errorf("could not convert block(%T) to evm.Block", stateBlock)
+		return fmt.Errorf("could not convert block(%T) to evm.EthBlockWrapper", stateBlock)
 	}
 
-	block := evmBlock.ethBlock
+	block := evmBlockGetter.GetEthBlock()
```

Comment:
I suppose this change is in support of moving the sync client to its own package?